src/main/java/windycall/Parser.java

```diff
@@ -0,0 +1,106 @@
+package windycall;
+
+/**
+ * Deals with making sense of the user command by splitting
+ * user command and translation
+ */
+public class Parser {
+
+    public Parser() {
+
+    }
+
+    /**
+     * Returns an operation type representing what users want to do
+     *
+     * @param command command input by users
+     * @return a enumeration type representing a specific operation
+     */
+    public OperationType getOperationType(String command) {
+        if (command.equals("list")) return OperationType.LIST;
+        String[] parts = command.split(" ");
+        switch (parts[0]) {
+        case "mark":
+            return OperationType.MARK;
+            //break;
+        case "unmark":
+            return OperationType.UNMARK;
+            //break;
+        case "delete":
+            return OperationType.DELETE;
+            //break;
+        case "find":
+            return OperationType.FIND;
+            //break;
+        default:
+            return OperationType.ADDTASK;
+            //break;
+        }
+    }
+
+    /**
+     * Returns the translated index of mark operation while
+     * handling invalid input
+     *
+     * @param parts parts of String of user command split by space
+     * @return an index representing which task user want to mark
+     */
+    public int getMarkIndex(String[] parts) {
+        if (parts.length == 1) {
+            System.out.println("     You should input a number to mark/unmark a task");
+            return -1;
+        }
+        int num = -1;
+        try {
+            num = Integer.parseInt(parts[1]);
+        } catch (NumberFormatException e) {
+            Ui.space();
+            System.out.println("☹ OOPS!!! You should input a number");
+        }
+        return num;
+    }
+
+    public int getUnmarkIndex(String[] parts) {
+        if (parts.length == 1) {
+            System.out.println("     You should input a number to mark/unmark a task");
+            return -1;
+        }
+        int num = -1;
+        try {
+            num = Integer.parseInt(parts[1]);
+        } catch (NumberFormatException e) {
+            Ui.space();
+            System.out.println("☹ OOPS!!! You should input a number");
+        }
+        return num;
+    }
+
+    public int getDeleteIndex(String[] parts) {
+        if (parts.length == 1) {
+            System.out.println("     You should input a number to delete a task");
+            return -1;
+        }
+        int num = -1;
```

Comment:
Can we use a more specific name, e.g. `index`? Similarly in other variables relating to the index.